internal/app/attorney_store.go

```diff
@@ -29,24 +29,37 @@ func (s *attorneyStore) Create(ctx context.Context, lpaOwnerKey dynamo.LpaOwnerK
 	attorney := &actor.AttorneyProvidedDetails{
 		PK:                 dynamo.LpaKey(data.LpaID),
 		SK:                 dynamo.AttorneyKey(data.SessionID),
-		UID:                attorneyUID,
+		UID:                shareCode.ActorUID,
 		LpaID:              data.LpaID,
 		UpdatedAt:          s.now(),
-		IsReplacement:      isReplacement,
-		IsTrustCorporation: isTrustCorporation,
+		IsReplacement:      shareCode.IsReplacementAttorney,
+		IsTrustCorporation: shareCode.IsTrustCorporation,
 		Email:              email,
 	}
 
-	if err := s.dynamoClient.Create(ctx, attorney); err != nil {
+	marshalledAttorney, err := attributevalue.MarshalMap(attorney)
+	if err != nil {
 		return nil, err
 	}
-	if err := s.dynamoClient.Create(ctx, lpaLink{
+
+	marshalledLink, err := attributevalue.MarshalMap(lpaLink{
 		PK:        dynamo.LpaKey(data.LpaID),
 		SK:        dynamo.SubKey(data.SessionID),
-		DonorKey:  lpaOwnerKey,
+		DonorKey:  shareCode.LpaOwnerKey,
 		ActorType: actor.TypeAttorney,
 		UpdatedAt: s.now(),
-	}); err != nil {
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	transaction := dynamo.NewTransaction().
+		Put(marshalledAttorney).
```

Comment:
I like this, my only suggestion would be to push the `MarhsalMap` calls in to `Put`. You could use a pattern like https://go.dev/blog/errors-are-values to hold onto the errors, then return in `WriteTransaction`

Reply:
Ah cool, that's precisely what I was trying to avoid so thanks for the link 👍

Reply:
(and this will make testing so much easier)